web/source/kmwosk.js

```diff
@@ -920,20 +1026,41 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
     }
 
     /**
-     * Function     getlayerId
+     * Function     getLayerId
      * Scope        Private
      * @param       {number}      m     shift modifier code
      * @return      {string}            layer string from shift modifier code (desktop keyboards)
-     * Description  Get name of layer form code (Note reversal: ctrlalt, not altctrl)
+     * Description  Get name of layer from code, where the modifer order is determined by ascending bit-flag value.
      */
     osk.getLayerId = function(m)
     {
       var s='';
-      if(m == 0) return 'default';
-      if(m & 1) s = 'shift';
-      if(m & 4) s = 'alt'+s;
-      if(m & 2) s = 'ctrl'+s;
-      return s;
+      if(m == 0) {
+        return 'default';
+      } else {
+        if(m & osk.modifierCodes.LCTRL) {
+          s = (s.length > 0 ? s + '-' : '') + 'leftctrl';
+        }
+        if(m & osk.modifierCodes.RCTRL) {
+          s = (s.length > 0 ? s + '-' : '') + 'rightctrl';
+        }
+        if(m & osk.modifierCodes.LALT) {
+          s = (s.length > 0 ? s + '-' : '') + 'leftalt';
+        }
+        if(m & osk.modifierCodes.RALT) {
+          s = (s.length > 0 ? s + '-' : '') + 'rightalt';
+        }
+        if(m & osk.modifierCodes.SHIFT) {
+          s = (s.length > 0 ? s + '-' : '') + 'shift';
+        }
+        if(m & osk.modifierCodes.CTRL) {
+          s = (s.length > 0 ? s + '-' : '') + 'ctrl';
+        }
+        if(m & osk.modifierCodes.ALT) {
+          s = (s.length > 0 ? s + '-' : '') + 'alt';
+        }
+        return s;
+      }
```

Comment:
Just wondering, do we support Caps and NCaps layers?

Reply:
Caps and NCaps layers are not default; they'd have to be implemented for Web by keyboard designers.  Their visual feedback is managed directly by toggling the OSK keys' visual properties.